presto-rcfile/src/main/java/com/facebook/presto/rcfile/GzipCodec.java

```
@@ -0,0 +1,344 @@
+import static java.lang.Math.toIntExact;
+
+public class GzipCodec
+        implements CompressionCodec
+{
+    private static final int GZIP_BUFFER_SIZE = toIntExact((new DataSize(8, KILOBYTE)).toBytes());
+
+    @Override
+    public CompressionOutputStream createOutputStream(OutputStream outputStream)
+            throws IOException
+    {
+        return new HadoopGzipOutputStream(outputStream, GZIP_BUFFER_SIZE);
+    }
+
+    @Override
+    public CompressionOutputStream createOutputStream(OutputStream outputStream, Compressor compressor)
+            throws IOException
+    {
+        if (!(compressor instanceof GzipCodec.HadoopGzipCompressor)) {
+            throw new IllegalArgumentException("Compressor is not the Gzip decompressor");
+        }
+        return new HadoopGzipOutputStream(outputStream, GZIP_BUFFER_SIZE);
+    }
+
+    @Override
+    public Class<? extends Compressor> getCompressorType()
+    {
+        return HadoopGzipCompressor.class;
+    }
+
+    @Override
+    public Compressor createCompressor()
+    {
+        return new HadoopGzipCompressor();
+    }
+
+    @Override
+    public CompressionInputStream createInputStream(InputStream inputStream)
+            throws IOException
+    {
+        return new HadoopGzipInputStream(inputStream, GZIP_BUFFER_SIZE);
+    }
+
+    @Override
+    public CompressionInputStream createInputStream(InputStream inputStream, Decompressor decompressor)
+            throws IOException
+    {
+        if (!(decompressor instanceof GzipCodec.HadoopGzipDecompressor)) {
+            throw new IllegalArgumentException("Decompressor is not the Gzip decompressor");
+        }
+        return new HadoopGzipInputStream(inputStream, GZIP_BUFFER_SIZE);
+    }
+
+    @Override
+    public Class<? extends Decompressor> getDecompressorType()
+    {
+        return HadoopGzipDecompressor.class;
+    }
+
+    @Override
+    public Decompressor createDecompressor()
+    {
+        return new HadoopGzipDecompressor();
```

If it is indeed unused, can we make this throw `UnsupportedOperationException`?